dvc/remote/base.py

```diff
@@ -912,20 +926,21 @@ def _cache_exists_traverse(
             ]
         with Tqdm(
             desc="Querying "
-            + ("cache in " + name if name else "remote cache"),
-            total=len(traverse_prefixes),
-            unit="dir",
+            + ("cache in '{}'".format(name) if name else "remote cache"),
+            total=remote_size,
+            initial=len(remote_checksums),
```

Comment:
Some checksums are fetched/processed during the size estimation step, and then are passed into this function as `remote_checksums`. So we are only processing the remaining `remote_size - len(remote_checksums)` checksums

Reply:
Did those `initial` things get a progressbar earlier? If yes, then this is fine. If not (because they were super quick) then we should probably instead just use:
```suggestion
            total=remote_size - len(remote_checksums),
```

Reply:
For the size estimation step, we fetch `initial` items, to get the estimated total (len(initial) * 4096`) and display the status of that estimation with the no-total progressbar. And then here we are displaying the status of fetching the remaining items out of the estimated total